src/vscode-bicep/src/commands/commandManager.ts

```diff
@@ -21,8 +29,42 @@ export class CommandManager extends Disposable {
   }
 
   private registerCommand<T extends Command>(command: T): void {
-    this.register(
-      vscode.commands.registerCommand(command.id, command.execute, command)
+    this.validateCommand(command);
+
+    // The command will be added to the extension's subscriptions and therefore disposed automatically
+    // when the extension is disposed.
+    azureextensionui.registerCommand(
+      command.id,
+      async (context: azureextensionui.IActionContext, ...args: unknown[]) => {
+        return await command.execute(context, ...args);
+      }
+    );
+  }
+
+  private validateCommand<T extends Command>(command: T): void {
```

Comment:
The is very helpful! However, I wonder if we could avoid validating each command when activating the extension during runtime. Could this be implemented as a unit test instead?

Reply:
I'm not sure that it can - we don't have a list of IDs that we can retrieve without registerCommand getting called.

Reply:
Makes sense.

Reply:
Btw, I realized that it's still possible to do that if we make the `id` fields static. It's a little hacky tbh, but I'm sharing the code anyway in case people are curious about it:

```TypeScript
import * as commands from "./command";


for (const command of Object.values(commands)) {
  if (!command.prototype) {
    continue;
  }

  var ownPropertyNames = Object.getOwnPropertyNames(command.prototype);

  if (
    ownPropertyNames.indexOf("constructor") >= 0 &&
    ownPropertyNames.indexOf("execute") >= 0 &&
    command.id &&
    typeof command.id === 'string'
  ) {
    console.log(command.id); // <== "bicep.showSource", "bicep.showVisualizer", ...
  }
}

```